Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -313,9 +324,15 @@ internal static string DeterminePrimaryScriptFile(JObject functionConfig, string
                         Path.GetFileNameWithoutExtension(p).ToLowerInvariant() == "run" ||
                         Path.GetFileName(p).ToLowerInvariant() == "index.js");
                 }
+            }
 
-                return functionPrimary;
+            if (string.IsNullOrEmpty(functionPrimary))
+            {
+                throw new ConfigurationErrorsException("Unable to determine the primary function script. Try renaming your entry point script to 'run' (or 'index' in the case of Node), " +
+                    "or alternatively you can specify the name of the entry point script explicitly by adding a 'scriptFile' property to your function metadata.");
```

Comment:
if we return null, exception messages will be lost (which are useful information for users, whereas currently if we give an invalid url, here's the message: `"Error:We are unable to reach your function app (Not Found). Please try again later."`), maybe we can do one of the following:
1. return the error string in `ScriptHref`
2. return null, add an error field to `functionEnvelop`
3. write the exception message to a error file in function folder, and pass its uri

I personally prefer **3**, as it requires almost no change on portal, but I don't know if it can cause function runtime/(3rd party which calls `api/functions`) to behave weirdly?